templates/pkg/resource/descriptor.go.tpl

```diff
@@ -172,3 +173,21 @@ func (d *resourceDescriptor) MarkUnmanaged(
 	}
 	k8sctrlutil.RemoveFinalizer(obj, finalizerString)
 }
+
+// MarkAdopted places descriptors on the custom resource that indicate the
+// resource was not created from within ACK.
+func (d *resourceDescriptor) MarkAdopted(
+	res acktypes.AWSResource,
+) {
+	obj := res.RuntimeMetaObject()
+	if obj == nil {
+		// Should not happen. If it does, there is a bug in the code
+		panic("nil RuntimeMetaObject in AWSResource")
+	}
+	curr := obj.GetAnnotations()
+	if curr == nil {
+		curr = make(map[string]string)
+	}
+	curr[ackv1alpha1.AnnotationAdopted] = "true"
+	obj.SetAnnotations(curr)
+} 
```

Comment:
I wonder if we'll need a `MarkUnadopted` ever? Probably not, and the user can manually remove the annotation if they ever really want to....